src/west/util.py

```diff
@@ -91,6 +91,30 @@ class WestNotFound(RuntimeError):
     '''Neither the current directory nor any parent has a West installation.'''
 
 
+def get_pwd_or_cwd():
+    '''To align with shells that support symbolic links, prefer the PWD
+    environment variable, otherwise return os.getcwd()
+    '''
+    pwd = os.environ.get('PWD', None)
+    cwd = os.getcwd()
+
+    if pwd is None:
+        return cwd
+
+    # west_topdir is crucial, so let's be paranoid.
```

Comment:
Yes, that would be relying on `/usr/bin/pwd -L` to perform this sanity check which seems to be true on the two operating systems I tried `PWD=fubar /usr/bin/pwd -L`[*]. Not sure such a sanity check can be taken for granted but to be honest this first check is totally paranoid; only in case someone evil corrupts PWD?

[*] interestingly, you can successfully fool `-L` like this if PWD is a valid match.